test/unittest/compiler/unittest_realizer.cpp

```diff
@@ -198,12 +197,51 @@ TEST(InputRealizer, remap_p) {
   realizeAndEqual(r, before, after);
 }
 
-TEST(PreviousInputRealizer, previous_p) {
-  {
-    std::vector<LayerRepresentation> before = {};
-
-    std::vector<LayerRepresentation> after = {};
+TEST(PreviousInputRealizer, DISABLED_previous_p) {
+  { /// realization without identifying custom input
+    std::vector<LayerRepresentation> before = {
+      {"fully_connected", {"name=fc1", "input_shape=1"}}, // model input
+      {"fully_connected", {"name=fc2"}}, // auto connected to fc 1
+      {"fully_connected", {"name=fc3", "input_layers=fc1"}},
+      {"fully_connected", {"name=fc4"}}, // auto connected to fc 3
+    };
+
+    std::vector<LayerRepresentation> after = {
+      {"fully_connected", {"name=fc1", "input_shape=1"}},
+      {"fully_connected", {"name=fc2", "input_layers=fc1"}},
+      {"fully_connected", {"name=fc3", "input_layers=fc1"}},
+      {"fully_connected", {"name=fc4", "input_layers=fc3"}},
+    };
     PreviousInputRealizer r({});
     realizeAndEqual(r, before, after);
   }
+  { /// realization identifying fc1, fc4 is input layer
+    std::vector<LayerRepresentation> before = {
+      {"fully_connected", {"name=fc1"}}, // will be identified as model input
+      {"fully_connected", {"name=fc2"}}, // auto connected to fc 1
+      {"fully_connected", {"name=fc3", "input_layers=fc1"}},
+      {"fully_connected", {"name=fc4"}}, // will be identified as model input
```

Comment:
add input_shape to be identified as model input?

Reply:
aah, its specified as a model input to the realizer in its arguments.
Consider adding a unittest where its not specified as a argument to the realizer and relies on input_shape to be interpreted as mode input.

Reply:
This is covered at line 203

Reply:
That is the first layer itself. New code doesn't take first layer into decision making, but old code did.
Hence the test case of input_shape for an intermediate layer :)

Reply:
Alright, I'll add

Reply:
Added